### PR TITLE
Add support for creating groups

### LINKED ISSUE #7: Create group
`GroupManager#create(options: GroupCreateOptions): Promise<Group>`

Creates a new group.
It would probably be worth describing the `GroupCreateOptions` object, but I'll decide on that later.

Usage:
```js
client.groups.create({ name: "my awesome group" })
```

### PR DESCRIPTION
**Pull request by striker4150**
Resolves #7.

### CODE REVIEW

**Comment by thewilloftheshadow** [comment]
Test failure was due to the fact it was running on a fork with no access to the testing account

**Comment by not-so-smart** [comment]
> Test failure was due to the fact it was running on a fork with no access to the testing account

yeah lol this is a bit of a problem, I feel like we should make it so that either the tests don't run on forks at all, or make it so that org members can manually run the tests on a pr after checking out the code if that's possible

**Review by not-so-smart — CHANGES_REQUESTED** [review]
Thanks for the PR! See comments, and remember to add `PostGroupBody` and `PostGroupResponse` to the type import from `groupme-api-types` on line 1.

**Comment by striker4150** [comment]
I made the changes as requested. Let me know if you have any other suggestions.

File: src/managers/GroupManager.ts [hunk 1]
```diff
@@ -50,7 +54,18 @@ export default class GroupManager extends BaseManager<Group> implements GroupMan
...
+        if (options.description !== undefined) reqBody.description = options.description
+        if (options.image_url !== undefined) reqBody.image_url = options.description
+        if (options.share !== undefined) reqBody.share = options.share
+        if (options.join_question !== undefined) {
+            reqBody.show_join_question = true
+            reqBody.join_question = { text: options.join_question, type: 'join_reason/questions/text' }
+        }
+        if (options.requires_approval !== undefined) reqBody.requires_approval = options.requires_approval
+        if (options.office_mode !== undefined) reqBody.office_mode = options.office_mode
+        const res = await this.client.rest.api<PostGroupResponse>('POST', 'groups', { body: reqBody })
```
Comment (not-so-smart) [inline src/managers/GroupManager.ts:68]: can you rename `reqBody` to just `body` and then use shorthand notation here? just for consistency with implementations of other API methods
```suggestion
        const res = await this.client.rest.api<PostGroupResponse>('POST', 'groups', { body })
```

**Review by not-so-smart — CHANGES_REQUESTED** [review]
looks great! one more tiny code quality/formatting change just for consistency and I think this is ready to merge 👍

**Review by not-so-smart — APPROVED** [review]
Looks good! Thanks for contributing 😊👍

**Comment by not-so-smart** [comment]
before we merge this, @thewilloftheshadow can you double check why the tests failed? I approved the actions to run but either the auth token didn't get shared or some test genuinely didn't pass. I also couldn't access the logs from this run for some reason, maybe you can get them somehow.

**Comment by thewilloftheshadow** [comment]
Yeah I can see the logs, it's from how I have them setup, I need to change some things. You can go ahead and merge

**Pull request merged**